applications/addons/controllers/class.addoncontroller.php

```diff
@@ -489,48 +489,123 @@ public function Approve($AddonID = '', $AddonVersionID = '') {
       }
       $VersionModel = new Gdn_Model('AddonVersion');
       $AddonVersion = $VersionModel->GetID($AddonVersionID, DATASET_TYPE_ARRAY);
-      
+
       if (!$AddonVersion['DateReviewed']) {
          $VersionModel->Save(array('AddonVersionID' => $AddonVersionID, 'DateReviewed' => Gdn_Format::ToDateTime()));
       } else {
          $VersionModel->Update(array('DateReviewed' => null), array('AddonVersionID' => $AddonVersionID));
       }
-      
+
       Redirect('/addon/'.AddonModel::Slug($Addon));
   }
-  
-   protected static function NotFoundString($Code, $Item) {
+
+  public function AttachToDiscussion($DiscussionID = NULL) {
+    $this->Permission('Addons.Addon.Manage');
+    $DiscussionModel = new DiscussionModel();
+    $Discussion = $DiscussionModel->GetID($DiscussionID);
+    if ($Discussion) {
+      $Addon = $this->AddonModel->GetID($Discussion->AddonID);
+      $this->Form->SetData($Addon);
+      $RedirectUrl = 'discussion/' . $Discussion->DiscussionID;
+    } else {
+      throw NotFoundException('Discussion');
+    }
+    
+    if ($this->Form->IsPostBack()) {
+      // Look up for an existing addon
+      $FormValues = $this->Form->FormValues();
+      $Addon = FALSE;
```

Comment:
Not gonna stop a PR over it, but we're switching to lowercase bools & null everywhere.